app/lib/features/segment_deletion_config.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Features
+  class SegmentDeletionConfig
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(config = {})
+      format_config(config)
+      check_validity(config)
+    end
+
+    def self.valid_environment?
+      # This is because this feature destroys real data unless the call to segment is stubbed and without real credentials in the config (test environment),
+      # so we prevent this feature to be used in preview or development thinking that t be just checking around while actually destroying real data by accident
+      env = Rails.env
+      env.production? || env.test?
```

Comment:
Cannot test in `preview`.

Reply:
Right @guicassolato , intentionally. Because as it says in the comment, if I enable it in preview, we will use it like if we were just 'testing' and we will be destroying REAL user data in Segment.